tests/foreman/api/test_hostgroup.py

```diff
@@ -148,6 +148,62 @@ def test_bz_1107708(self):
         self.assertEqual(host_attrs['all_puppetclasses'][0]['name'], 'ntp')
 
 
+@skip_if_bug_open('bugzilla', 1235377)
+class MissingAttrTestCase(APITestCase):
+    """Tests to see if the server returns the attributes it should."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a ``HostGroup`` as ``cls.host_group``."""
+        host_group = entities.HostGroup().create()
+        cls.hg_attrs = host_group.read_json()
+
+    def test_get_content_source(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains some value for the ``content_source``
+        field.
+
+        @Feature: HostGroup
+
+        """
+        attrs = ('content_source', 'content_source_id')
+        self.assertTrue(
+            any([attr in self.hg_attrs for attr in attrs]),
+            'None of {0} are in {1}'.format(attrs, self.hg_attrs.keys())
+        )
+
+    def test_get_content_view(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains either a ``content_view`` or
+        ``content_view_id`` attribute.
+
+        @Feature: HostGroup
+
+        """
+        attrs = ('content_view', 'content_view_id')
+        self.assertTrue(
+            any([attr in self.hg_attrs for attr in attrs]),
+            'None of {0} are in {1}'.format(attrs, self.hg_attrs.keys())
+        )
+
+    def test_get_lifecycle_environment(self):
+        """@Test: Read a host group. Inspect the server's response.
+
+        @Assert: The response contains either a ``lifecycle_environment`` or
+        ``lifecycle_environment_id`` attribute.
+
+        @Feature: HostGroup
+
+        """
+        attrs = ('lifecycle_environment', 'lifecycle_environment_id')
+        self.assertTrue(
+            any([attr in self.hg_attrs for attr in attrs]),
+            'None of {0} are in {1}'.format(attrs, self.hg_attrs.keys())
+        )
```

Comment:
And yes, I realize that we now have identical-looking code. That's fine. Let's just add unittest2 as a dependency (and drop DDT), and we can use the `subTest` context manager:

```python
for attr in ('lifecycle_environment', 'content_view', 'content_source'):
    with self.subTest(attr):
         _get_entity_id(attr, self.host_group_attrs)
```